addons/web/static/src/js/views/basic/basic_model.js

```diff
@@ -3285,6 +3285,22 @@ var BasicModel = AbstractModel.extend({
         }
         return commands;
     },
+    /**
+     * For a given resource of type 'record', get the active field, if any.
+     *
+     * Since the ORM can support both `active` and `x_active` fields for
+     * the archiving mechanism, check if any such field exist and prioritize
+     * them. The `active` field should always take priority over its custom
+     * version.
+     *
+     * @param {Object} record local resource
+     * @returns {String|undefined} the field name to use for archiving purposes
+     *   ('active', 'x_active') or undefined if no such field is present
+     */
+    _getActiveField: function(record) {
+        const fields = Object.keys(record.fields);
+        return fields.find((f) => f === 'active' || f === 'x_active');
```

Comment:
What if fields contains both `active` and `x_active`?
The doc says it should return `active`, but the implementation does not seem deterministic.

Reply:
By the way, using `find` seems inefficient to me. I suggest
```suggestion
        return ('active' in fields ? 'active' :
                'x_active' in fields ? 'x_active' :
                undefined);
```